src/python_pachyderm/client.py

```diff
@@ -172,80 +194,157 @@ def new_from_pachd_address(
             root_certs=root_certs,
             transaction_id=transaction_id,
             tls=u.scheme == "grpcs" or u.scheme == "https",
+            use_default_host=False,
         )
 
     @classmethod
-    def new_from_config(cls, config_file=None):
+    def new_from_config(cls, config_file):
         """
-        Creates a Pachyderm client from a config file, which can either be
-        passed in as a file-like object, or if unset, checks the PACH_CONFIG env
-        var for a path. If that's also unset, it defaults to loading from
-        '~/.pachyderm/config.json'.
+        Creates a Pachyderm client from a config file-like object.
 
         Params:
 
-        * `config_file`: An optional file-like object containing the config
-        json file. If unspecified, we load the config from the default
-        location ('~/.pachyderm/config.json'.)
+        * `config_file`: A file-like object containing the config
+        json file.
         """
 
-        if config_file is not None:
-            j = json.load(config_file)
-        elif "PACH_CONFIG" in os.environ:
-            with open(os.environ.get("PACH_CONFIG"), "r") as config_file:
+        if config_file is None:
+            raise ConfigError("no config object provided")
+
+        config = json.load(config_file)
+        (
+            _,
+            _,
+            pachd_address,
+            auth_token,
+            root_certs,
+            transaction_id,
+            _,
+        ) = cls._parse_config(config)
+
+        client = cls.new_from_pachd_address(
+            pachd_address,
+            auth_token=auth_token,
+            root_certs=root_certs,
+            transaction_id=transaction_id,
+        )
+
+        context = cls._get_active_context(config)
+        expected_deployment_id = context.get("cluster_deployment_id")
+        if expected_deployment_id:
+            cluster_info = client.inspect_cluster()
+            if cluster_info.deployment_id != expected_deployment_id:
+                raise BadClusterDeploymentID(
+                    expected_deployment_id, cluster_info.deployment_id
+                )
+
+        return client
+
+    @staticmethod
+    def check_for_config():
+        """
+        Checks for config.
+
+        Returns:
+        * `j`: The config as a json object.
+        """
+
+        j = Client.check_pach_config_env_var()
```

Comment:
Do you know a slicker way to do these checks one by one?

Reply:
I think this is slick enough 😉
You could try something like

```python
configs = [Client.check_pach_config_env_var(), Client.check_pach_config_spout(), Client.check_pach_config_local()]
return next((config for config in configs if config), "boo no configs")
```